packages/react-dom/src/__tests__/validateDOMNesting-test.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+function normalizeCodeLocInfo(str) {
+  return str && str.replace(/at .+?:\d+/g, 'at **');
+}
+
+function expectInvalidNestingWarning(shouldWarn, tagsList, warningsList = []) {
+  let element = null;
+  let tags = tagsList;
```

Comment:
This doesn’t copy the array, it only creates a reference to it. Therefore the `splice` call later mutates it.

It would be clearer IMO if we didn’t mutate the input. Otherwise, for example, declaring an array as a variable in the test and passing it two times wouldn’t work. We don’t do this but it could be potentially confusing in the future.

To solve it we need to copy the array on this line.